### ISSUE TITLE
softmax hessian

### ISSUE THREAD

**Issue by benoitdescamps** [dmlc/xgboost#2485]
I am currently trying to implement a multivariant extension to xgboost.
Unless I am mistaken, the formula for the weights remain the same, except for the hessian being a matrix, so there should be a matrix inverse somewhere, if you go for multivariate.

In the current version of xgboost, there is however a softmax implementation. 
In the code,
I see in \src\objective\multiclass_obj.cc that you are just using the approximation p(1-p)*weight for the Hessian.
Is it documented anywhere that you are using this approximation, or I might be mistaken and this is the real Hessian?

**Comment by tqchen** [comment]
https://github.com/dmlc/xgboost/issues/1825

**Comment by benoitdescamps** [comment]
Thanks for the quick answer. The paper (http://homes.cs.washington.edu/~tqchen/data/pdf/GBCRF-AISTATS15.pdf) indeed answered my question!

**Comment by tqchen** [comment]
if you are interested, you are more than welcomed to contribute a tutorial in explaining the details:)
